src/selection_stream/selection_stream_processor.ts

```diff
@@ -338,13 +346,17 @@ export class SelectionStreamProcessor
 
     // The whole sheet is selected, select the anchor cell
     if (isEqual(this.anchor.zone, this.getters.getSheetZone(sheetId))) {
-      return this.selectZone({ ...anchor, zone: positionToZone(anchor.cell) });
+      return this.modifyAnchor(
+        { ...anchor, zone: positionToZone(anchor.cell) },
+        "overrideSelection",
+        "AlterZone"
+      );
     }
 
     const tableZone = this.expandZoneToTable(anchor.zone);
 
     return !deepEquals(tableZone, anchor.zone)
-      ? this.selectZone({ ...anchor, zone: tableZone })
+      ? this.modifyAnchor({ ...anchor, zone: tableZone }, "overrideSelection", "AlterZone")
```

Comment:
```suggestion
      ? this.modifyAnchor({ ...anchor, zone: tableZone }, "updateAnchor", "AlterZone")
```
the concept of altering a zone while overriding the zone makes no sense imho, this change does not seem to break anything